src/components/CohortBuilder/FieldFilterContainer.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+
+import Row from 'uikit/Row';
+import { Button } from 'antd';
+
+import './CohortBuilder.css';
+
+const FilterContainer = props => <div class="cb-filterContainer">{props.children}</div>;
+const Header = props => <div class="cb-filterContainer-header">{props.children}</div>;
+const Footer = props => <Row className="cb-filterContainer-footer">{props.children}</Row>;
+
+export const FieldFilterContainer = ({
+  children,
+  onSubmit: onSqonSubmit,
+  onCancel,
+  onBack,
+  applyEnabled = true,
+  showHeader = true,
+  className = '',
+}) => {
+  const [isDisabled, setDisabled] = useState(true);
+
+  const hasSelectedElements = handler => {
+    const checkboxes = handler.parentElement.parentElement.parentElement.querySelectorAll(
+      'input[type="checkbox"]:checked',
+    );
+    const toggles = handler.parentElement.parentElement.querySelectorAll(
+      'div.toggle-button-option.active',
+    );
+    let ranges = 0;
+    const rangeNodes = handler.parentElement.parentElement.parentElement.querySelectorAll(
+      'div.rangeInputContainer input.rangeFilterInput:not([disabled])',
+    );
+    rangeNodes.forEach((input, key) => {
+      const node = rangeNodes.item(key);
+      if (node.value.length > 0 && node.value > -1) {
+        ranges++;
+      }
+    });
+    return setDisabled(
+      checkboxes.length < 1 &&
+        toggles.length < 1 &&
+        (ranges < rangeNodes.length || rangeNodes.length === 0),
+    );
+  };
```

Comment:
It is just a question, maybe it d make things worst but could this be extracted outside de FieldFilterContainer?

const hasSelectedElements = (handler, setDisableCB) => {....setDisableCB(conds)}

Reply:
in another file or outside the funciton?

Reply:
outside the func if not needed elsewhere